src/datatypes/src/types/date_type.rs

```diff
@@ -59,6 +59,7 @@ impl LogicalPrimitiveType for DateType {
     type ArrowPrimitive = Date32Type;
     type Native = i32;
     type Wrapper = Date;
+    type LargestType = DateType;
```

Comment:
```suggestion
    type LargestType = Self;
```
Would this be more clear?

Reply:
As well as other types that cannot up-cast